src/System.Collections.Immutable/tests/ImmutableArrayBuilderTest.cs

```diff
@@ -288,10 +288,18 @@ public void Sort()
         [Fact]
         public void Sort_Comparison()
         {
-            var builder = new ImmutableArray<int>.Builder();
+            var builder = new ImmutableArray<int>.Builder(4);
```

Comment:
can you make these arrays of strings.  that way your comparer will null ref if you get passed a default element?